backup/tar_test.go

```diff
@@ -87,3 +90,34 @@ func TestCheckIfEmpty(t *testing.T) {
 	// Case 4: Passed a file instead of a directory
 	assert.Error(t, checkIfEmpty(myfile))
 }
+
+func TestPathTraversal(t *testing.T) {
+	// Create a malicious tarball that tries to write to a relative directory.
+	var tarball bytes.Buffer
+	tw := tar.NewWriter(&tarball)
+	data := []byte("something malicious")
+	hdr := &tar.Header{
+		Name: "../evil.txt",
```

Comment:
Poking around a bit at similar fixes, it seems like you can embed multiple ../ into the beginning of the file path, should we check for that as well? I'm not clear on whether you can embed ../ into the middle of the path, but might also be worth testing just to document that that case is handled as well (with or without the fix).

Reply:
Added more tests.